### PR TITLE
Fixed a typo bug.

### PR DESCRIPTION
**Pull request by dfoverdx**
Fixed a potential race condition bug.
Made window not be displayed until loading the page has finished.  This way, if the person is already logged in, the window does not appear at all.
Allowed passing in custom BrowserWindow intialization options.
Allowed passing in a custom BrowserWindow that may or may not be closed upon login.
Added option to close the app if the user closes the window before logging in.  Useful for apps where login is required up front.

### CODE REVIEW

File: packages/twitch-electron-auth-provider/src/index.ts [hunk 1]
```diff
@@ -1,17 +1,73 @@
...
+	exitAppWhenClosedAndNotDone: false,
+	escapeToClose: true,
+	closeOnLogin: true,
+};
+
 export default class ElectronAuthProvider implements AuthProvider {
 	private _accessToken?: AccessToken;
-	private readonly _currentScopes: Set<string> = new Set();
+	private readonly _currentScopes = new Set<string>();
+	private readonly _clientCredentials: TwitchClientCredentials;
```
[the remark this answers is not in the document]
Comment (dfoverdx) [reply packages/twitch-electron-auth-provider/src/index.ts]: Oh, right, it was a stylistic thing.  I hadn't originally intended to PR this, and I didn't like `_` showing up in my IntelliSense.  Should I change it back?

File: packages/twitch-electron-auth-provider/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "twitch-electron-auth-provider",
-  "version": "3.0.1",
+  "version": "3.0.2",
```
[the remark this answers is not in the document]
Comment (dfoverdx) [reply packages/twitch-electron-auth-provider/package.json]: Ah, good to know

File: packages/twitch-electron-auth-provider/tsconfig.json [hunk 3]
```diff
@@ -2,7 +2,8 @@
 	"extends": "../../tsconfig",
 	"compilerOptions": {
 		"rootDir": "src",
-		"outDir": "lib"
+		"outDir": "lib",
+		"sourceMap": true,
```
[the remark this answers is not in the document]
Comment (dfoverdx) [reply packages/twitch-electron-auth-provider/tsconfig.json]: Oh, weird, for some reason I wasn't seeing it be generated.  I'll undo this.

File: packages/twitch-electron-auth-provider/src/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (d-fischer) [reply packages/twitch-electron-auth-provider/src/index.ts]: Yes please.

File: packages/twitch-electron-auth-provider/tsconfig.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
 	"extends": "../../tsconfig",
 	"compilerOptions": {
 		"rootDir": "src",
-		"outDir": "lib"
+		"outDir": "lib",
```
Comment (d-fischer) [inline packages/twitch-electron-auth-provider/tsconfig.json]: This is invalid JSON. Even though TS tolerates this, I would prefer if it wasn't.

Comment (dfoverdx) [reply packages/twitch-electron-auth-provider/tsconfig.json]: Fixed

**Pull request merged**